x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/select_rule_type/index.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+import { EuiCard, EuiFlexGrid, EuiFlexItem, EuiIcon, EuiFormRow } from '@elastic/eui';
+
+import { FieldHook } from '../../../../../shared_imports';
+import { RuleType } from '../../../../../containers/detection_engine/rules/types';
+import * as i18n from './translations';
+import { isMlRule } from '../../helpers';
+
+interface SelectRuleTypeProps {
+  field: FieldHook;
+}
+
+const Wrapper = styled(EuiFormRow)``;
+
+export const SelectRuleType = ({ field }: SelectRuleTypeProps) => {
+  const ruleType = field.value as RuleType;
+  const setType = useCallback(
+    (type: RuleType) => {
+      field.setValue(type);
+    },
+    [field]
+  );
+  const setMl = useCallback(() => setType('machine_learning'), [setType]);
+  const setQuery = useCallback(() => setType('query'), [setType]);
+  const license = true; // TODO
```

Comment:
👍  I can help you with this if you have questions. Appreciate the ahead of time license check. Up to you if you want to write a separate ticket for adding the license check so we don't forget about it.

Reply:
It's next on my list in #60301